site/src/pages/release-notes/1.9.0.mdx

```diff
@@ -0,0 +1,237 @@
+- [TCP User Timeout Option](https://datatracker.ietf.org/doc/html/rfc5482) is enabled by default when an idle
+  timeout is enabled. #3509
+  - `TCP_KEEPIDLE` and `TCP_KEEPINTVL` options are enabled when a PING interval is greater than 0.
+- You can now get the last value of a header name or a query param using
+  <type://HttpHeaders#getLast(CharSequence)> and <type://QueryParams#getLast(String)>. #3438 #3568
+  ```java
+  HttpHeaders headers = ...;
+  headers.getLast(HttpHeaderNames.X_FORWARDED_FOR);
+  
+  QueryParams params = ...;
+  params.getLast("items");
+  ```
+- You can now see the specific information such as the length of transferred, delta, and limit
+  from a <type://ContentTooLargeException>. #3520 #3616
+- You can now convert a <type://Multipart> into an <type://HttpResponse>. #3642
+- You can now warm up an `EventLoopGroup` using <type://EventLoopGroups#warmUp(EventLoopGroup)>. #3363 #3610
+- You can now figure out how many connections that a <type://ClientFactory> is managing
+  using <type://ClientFactory#numConnections()>. #3596 #3613
+- You can now specify the number of event loop threads and blocking task threads. #3597 #3602
+  ```java
+  Server.builder()
+        .workerGroup(3)
+        .blockingTaskExecutor(10)
+        ...
+        .build();
+  
+  ClientFactory.builder()
+               .workerGroup(3)
+               .build();
+  ```
+- You can now clear the mocking state of a <type://MockWebServerExtension> using
+  <type://MockWebServerExtension#reset()>. #3629
+
+## 📈 Improvements
+
+- Various optimizations for <type://StreamMessage> implementations. #3585 #3584
+- You can now use an optimized context-aware `CompletableFuture` for Java 12 and above. #2119 #3615
+- Various multipart mime types are added to <type://MediaType>. #3642
+- `TemporaryThreadLocals` throws an `IllegalStateException` if it was not properly released before using it.
+  #3505
+
+## 🛠️ Bug fixes
+
+- `ByteBuf`s are no longer leaked when using:
+  - Spring WebFlux integration with compression #3608 #3617
+  - <type://StreamMessageDuplicator> #3476 #3583
+- You no longer see a <type://ResponseTimeoutException> with <type://CircuitBreakerClient> when a large number
+  of messages are received from a response. #3631
+- You can now set the service name of an annotated service using
+  the <type://ServerBuilder#defaultServiceNaming(ServiceNaming)>. #3589
+- You no longer see a <type://ClosedStreamException> when an <type://EndpointGroup> is empty. #3381 #3636
+- Late `Subscriber`s correctly run on the specified executor. #3623 #3632
+- An exception raised in <type://THttpService> is now properly peeled using
+  <type://Exceptions#peel(Throwable)>. #3588
+- You no longer see an `UnsatisfiedLinkError` when a Netty JAR without native libraries is included. #3485 #3607
+- Additional headers are now correctly updated to <type://RequestOnlyLog#requestHeaders()>. #3579 #3581
+- `CancellationScheduler` provides more accurate timeout nanoseconds. #3570
```

Comment:
Could make this more user-friendly. A user has no idea what `CancellationScheduler`.